### ISSUE TITLE
Idempotency causing hang

### ISSUE THREAD

**Issue by SturmB** [ansible/ansible#78903]
### Summary

When I attempt to run my Ansible playbook on Pop!_OS (based on Ubuntu), I am running into an issue where a previously-run task now hangs. There is no indication of an error nor a message that explains what's going on.

It started during an `apt` task. When it started hanging upon subsequent runs of the playbook, I added the `never` tag to it so it could get past it. It did, then when I got an error in a later task, after I corrected the role that caused the error, running the playbook again caused it to hang on the task after the `apt` one. Setting it to `never`, the next one hangs. Then the next. Etc.

### Issue Type

Bug Report

### Component Name

apt

### Ansible Version

```console
$ ansible --version
ansible [core 2.13.4]
  config file = None
  configured module search path = ['/home/kerban/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/kerban/.local/lib/python3.10/site-packages/ansible
  ansible collection location = /home/kerban/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/kerban/.local/bin/ansible
  python version = 3.10.6 (main, Aug 10 2022, 11:40:04) [GCC 11.3.0]
  jinja version = 3.0.3
  libyaml = True
```

### Configuration

```console
# if using a version older than ansible-core 2.12 you should omit the '-t all'
$ ansible-config dump --only-changed -t all
DEFAULT_HOST_LIST(/home/kerban/ansible-dotfiles/ansible.cfg) = ['/home/kerban/a>
DEFAULT_LOAD_CALLBACK_PLUGINS(/home/kerban/ansible-dotfiles/ansible.cfg) = True
DEFAULT_ROLES_PATH(/home/kerban/ansible-dotfiles/ansible.cfg) = ['/home/kerban/>
DEFAULT_STDOUT_CALLBACK(/home/kerban/ansible-dotfiles/ansible.cfg) = yaml
DEFAULT_VAULT_PASSWORD_FILE(/home/kerban/ansible-dotfiles/ansible.cfg) = /home/>
HOST_KEY_CHECKING(/home/kerban/ansible-dotfiles/ansible.cfg) = False
INTERPRETER_PYTHON(/home/kerban/ansible-dotfiles/ansible.cfg) = /usr/bin/python3

CONNECTION:
==========

paramiko_ssh:
____________
host_key_checking(/home/kerban/ansible-dotfiles/ansible.cfg) = False

ssh:
___
host_key_checking(/home/kerban/ansible-dotfiles/ansible.cfg) = False
```

### OS / Environment

Pop!_OS 22.04 LTS

### Steps to Reproduce

The main playbook used is `playbooks/prephome.yml` in my [ansible-dotfiles repo](https://github.com/SturmB/ansible-dotfiles). It is executed with

```sh
ansible-playbook -K ./playbooks/prephome.yml --limit local
```

The first task where Ansible hung:
```yaml (paste below)
    - name: Ensure apt packages are installed
      ansible.builtin.apt:
        state: latest
        update_cache: yes
        # upgrade: yes
        name:
          - cowsay
          - fortune
          - figlet
          - neofetch
          - bat
          - php-gd
          - php-zip
          - tree
      become: yes
```

### Expected Results

I expect the playbook to run without hanging at any step.

### Actual Results

```console
<localhost> ESTABLISH LOCAL CONNECTION FOR USER: kerban
<localhost> EXEC /bin/sh -c 'echo ~kerban && sleep 0'
<localhost> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/kerban/.ansible/tmp `"&& mkdir "` echo /home/kerban/.ansible/tmp/ansible-tmp-1664322891.0145888-17202-72281950657863 `" && echo ansible-tmp-1664322891.0145888-17202-72281950657863="` echo /home/kerban/.ansible/tmp/ansible-tmp-1664322891.0145888-17202-72281950657863 `" ) && sleep 0'
Using module file /home/kerban/.local/lib/python3.10/site-packages/ansible/modules/apt.py
<localhost> PUT /home/kerban/.ansible/tmp/ansible-local-171153twfinqd/tmpw6nkgm36 TO /home/kerban/.ansible/tmp/ansible-tmp-1664322891.0145888-17202-72281950657863/AnsiballZ_apt.py
<localhost> EXEC /bin/sh -c 'chmod u+x /home/kerban/.ansible/tmp/ansible-tmp-1664322891.0145888-17202-72281950657863/ /home/kerban/.ansible/tmp/ansible-tmp-1664322891.0145888-17202-72281950657863/AnsiballZ_apt.py && sleep 0'
<localhost> EXEC /bin/sh -c 'sudo -H -S -p "[sudo via ansible, key=weujdtttruigpycytbxinybuxapjzrve] password:" -u root /bin/sh -c '"'"'echo BECOME-SUCCESS-weujdtttruigpycytbxinybuxapjzrve ; /usr/bin/python3 /home/kerban/.ansible/tmp/ansible-tmp-1664322891.0145888-17202-72281950657863/AnsiballZ_apt.py'"'"' && sleep 0'
```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

**Comment by ansibot** [comment]
Files identified in the description:
* [`lib/ansible/modules/apt.py`](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/apt.py)

If these files are incorrect, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/devel/ISSUE_HELP.md)

**Comment by sivel** [comment]
This more than likely sounds like a package is asking some question.

However, since I cannot replicate this, debugging the cause is going to be up to you.

I've written up some instructions on how you can debug and determine what code is hanging, and what command is causing it.

https://gist.github.com/sivel/5076af99d310dbda3188722f02cddeb4

Please provide the output from the debugging indicated in that gist.

needs_info

**Comment by SturmB** [comment]
Took me a while and many trial-and-error runs, but I found the problem, but I'm unsure how to work around it.

The playbook in question runs locally rather than connecting to remote servers. Also, I set up my local Pop!OS installation so that whenever I run a terminal command that requires elevation (i.e., `sudo ...`), it prompts for a fingerprint scan instead of asking for the password. This is not a problem when I run a simple sudo command, but can present a problem in an Ansible playbook that requires elevation. Even if I run the playbook with `-K` and give it my password, it still pauses when it gets to the `ansible.builtin.apt` task (the first one in my playbook that requires elevation).

I can temporarily fix this issue by running something like `sudo apt update` and then immediately run the playbook so that the timer for my fingerprint scan hasn't expired by the time it gets to the `become` tasks. I'm sure there's a more elegant Ansible solution, but I do not know what that solution might be.
